### DISCUSSION TITLE
Fast refresh / Hot reload does not work right away - takes minutes to kick in

### DISCUSSION THREAD

**Discussion by nodeworks** [vercel/next.js#17157] [Help] [answered]
I've been using next.js for a couple years and i'd say the past year or so i've had this problem of HMR and more recently fast refresh. When I load a page in the browser it doesn't hook up to fast refresh right away. If I make changes it doesn't do anything for about the first 5 minutes. Then randomly it will kick in at some point. How can I start with debugging this?

NextJS: 9.5.3
Docker for Mac: 2.3.6.1

**Comment by nodeworks** [comment]
Updated screenshots:

This is when I load the page and it doesn't work for a while (minutes):

![Screen Shot 2020-09-16 at 9 06 45 PM](https://user-images.githubusercontent.com/5272855/93408645-11110400-f863-11ea-8040-d3eacdba5471.png)

This is when it starts working and kicking in:

![Screen Shot 2020-09-16 at 9 09 46 PM](https://user-images.githubusercontent.com/5272855/93408632-05bdd880-f863-11ea-959e-bee58feff0ee.png)

**Comment by timneutkens** [comment]
Do you have a leftover service-worker from another project maybe? That would generally cause issues as it does not know how to handle certain requests.

**Comment by nodeworks** [comment]
Just checked and there are no service works for the domain.

**Comment by nodeworks** [comment]
I think I may have actually tracked down the source. so I use traefik as a reverse proxy. I think that may be causing the issue. I'm going to dive into it more and see if I can pinpoint why and where.

**Comment by nodeworks** [comment, marked as answer]
The issue is related to compression:

```
traefik.http.routers.node.middlewares: compress_mid@docker
traefik.http.middlewares.compress_mid.compress: true
```

Having gzip compression enabled was causing the issue. For traefik that config is here (https://docs.traefik.io/middlewares/compress/).

Turning it off fixes the issue.
